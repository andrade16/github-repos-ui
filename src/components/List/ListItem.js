import React from "react";
import Avatar from "../Avatar/Avatar";
import "./ListItem.scss";

const ListItem = ({ title, subtitle, content, avatarSrc }) => {
  return (
    <li className="list-item">
      <div className="titles-container">
        <div className="avatar-title-container">
          <Avatar width={30} height={30} src={avatarSrc} />
          <div className="title">{title}</div>
        </div>
        <div className="subtitle">{subtitle}</div>
      </div>
      <div className="content-container">{content}</div>
    </li>
  );
};

export default ListItem;
