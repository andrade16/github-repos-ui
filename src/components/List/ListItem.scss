@import "src/utils/variables";

.list-item {
  border-bottom: $light-border;
  margin: 5px;

  .titles-container {
    display: $flex-display;
    flex-direction: row;
    justify-content: space-between;
    padding: 4px;

    .avatar-title-container {
      display: $flex-display;
      flex-direction: row;
      align-items: $content-center;
    }

    .title,
    .subtitle {
      margin: 5px;
    }

    .title {
      font-size: 14px;
      font-weight: bold;
    }

    .subtitle {
      font-size: 12px;
    }
  }

  .content-container {
    padding: 4px;
    font-size: 12px;
  }
}
