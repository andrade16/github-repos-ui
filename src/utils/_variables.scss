$blue: #007bff;
$white: #fff;
$red: #dc3545;
$small-border-radius: 0.25rem;
$large-border-radius: 50%;
$loader-borders: 10px solid rgba(136, 136, 136, 0.2);
$light-border: 1px solid rgba(0, 0, 0, 0.125);
$pointer-cursor: pointer;
