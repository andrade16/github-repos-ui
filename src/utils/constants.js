export const Y_OFFSET_THRESHOLD = 400;
