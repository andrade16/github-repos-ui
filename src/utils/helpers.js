import moment from "moment";

export function formatDateFromNow(date) {
  return moment.utc(date).local().fromNow();
}

export function formatDateByDay(date) {
  return moment.utc(date).local().format("MMM D, YYYY");
}
