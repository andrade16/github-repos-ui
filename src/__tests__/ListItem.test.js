import React from "react";
import renderer from "react-test-renderer";
import ListItem from "../components/List/ListItem";
import { formatDateFromNow } from "../utils/helpers";

const mockCommitObj = {
  sha: "950841c25ea005f30050cdac545f87933cea038a",
  author: {
    login: "nhcarrigan",
    avatar_url: "https://avatars.githubusercontent.com/u/63889819?v=4",
  },
  commit: {
    committer: { date: "2021-02-18T16:04:03Z" },
    message: "test",
  },
};

describe("<ListItem/>", () => {
  it("Renders <ListItem/> correctly", () => {
    const tree = renderer
      .create(
        <ListItem
          title={mockCommitObj.author.login}
          subtitle={`commited ${formatDateFromNow(
            mockCommitObj.commit.committer.date
          )}`}
          content={mockCommitObj.commit.message}
          avatarSrc={mockCommitObj.author.avatar_url}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
