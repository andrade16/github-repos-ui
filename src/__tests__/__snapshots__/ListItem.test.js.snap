// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem/> Renders <ListItem/> correctly 1`] = `
<li
  className="list-item"
>
  <div
    className="titles-container"
  >
    <div
      className="avatar-title-container"
    >
      <div
        className="avatar"
      >
        <img
          alt="avatar-image"
          className="avatar-image"
          src="https://avatars.githubusercontent.com/u/63889819?v=4"
          style={
            Object {
              "height": 30,
              "width": 30,
            }
          }
        />
      </div>
      <div
        className="title"
      >
        nhcarrigan
      </div>
    </div>
    <div
      className="subtitle"
    >
      commited 5 days ago
    </div>
  </div>
  <div
    className="content-container"
  >
    test
  </div>
</li>
`;
